### ISSUE TITLE
MD024/no-duplicate-heading siblings_only: true has no affect

### ISSUE THREAD

**Issue by seanthegeek** [DavidAnson/markdownlint#1882]
I created the following `.markdownlint.yml` file to enable the `simlings_only` option in  [MD024/no-duplicate-heading](https://github.com/DavidAnson/markdownlint/blob/v0.38.0/doc/md024.md)

```yml
no-duplicate-heading:
  siblings_only: true
```

When I create a changelog markdown file like in the example:

```markdown
# Change log

## 1.0.0

### Features

## 2.0.0

### Features
```

`markdownlint` still raises a warning:

> MD024/no-duplicate-heading: Multiple headings with the same content

I know that the `.markdownlint.yml` is being read, because the warning goes away if I disable the rule completely.

```yml
no-duplicate-heading: false
```

**Comment by DavidAnson** [comment]
I can only superficially check this from my phone, but it seems to be working as expected, so I suspect the issue is something to do with the configuration you're providing. I'd start by making sure that YAML file is being parsed successfully.

https://dlaa.me/markdownlint/#%25m%23%20Change%20log%0A%0A%23%23%201.0.0%0A%0A%23%23%23%20Features%0A%0A%23%23%202.0.0%0A%0A%23%23%23%20Features%0A%0A%3C!--%20markdownlint-configure-file%20%7B%0A%20%20%22no-duplicate-heading%22%3A%20%7B%0A%20%20%20%20%22siblings_only%22%3A%20true%0A%20%20%7D%0A%7D%20--%3E%0A

**Comment by seanthegeek** [comment]
It might be an issue with the implantation in Visual Studio Code then. I've tried both JSON and YANL config files with the same result.

**Comment by DavidAnson** [comment]
Here's a repository/branch containing exact copies of the two files you provide above. You can clone and open the folder in VS Code with the markdownlint extension to confirm that no violations are reported.

<https://github.com/DavidAnson/playground/tree/markdownlint-issue-1882>

Please feel free to PR a change to that repository to demonstrate the problem you're seeing.

**Comment by seanthegeek** [comment]
I'm an idiot. I should not open issues with little sleep.

https://github.com/domainaware/checkdmarc/commit/f550117e7dd589044af3f6f2ed6b55e5a48c12cf
